packages/grid/x-data-grid/src/tests/keyboard.DataGrid.test.tsx

```diff
@@ -313,20 +314,44 @@ describe('<DataGrid /> - Keyboard', () => {
       fireEvent.keyDown(document.activeElement!, { key: 'PageDown' });
       expect(getActiveCell()).to.equal(`5-1`);
     });
+
+    it('should be able to type in an child input', () => {
+      const columns = [
+        {
+          field: 'name',
+          headerName: 'Name',
+          width: 200,
+          renderHeader: () => <input type="text" data-testid="custom-input" />,
+        },
+      ];
+
+      const rows = [
+        {
+          id: 1,
+          name: 'John',
+        },
+      ];
+
+      render(
+        <div style={{ width: 300, height: 300 }}>
+          <DataGrid rows={rows} columns={columns} />
+        </div>,
+      );
+      const input = screen.getByTestId('custom-input');
+      input.focus();
+      userEvent.type(input, 'hello world');
```

Comment:
Is not `fireEvent.keyDown` is sufficient? I always challenged myself to not add `@testing-library/user-event`. Although it offers some handy helpers it's rare when we need to fire a full click/keyboard event.

Reply:
Testing with 
```js
const input = screen.getByTestId('custom-input');
input.focus();
fireEvent.keyDown(input, {
key: "h",
code: "KeyH"
});
expect(input.value).to.equal('h');
```
leads to `AssertionError: expected '' to equal 'h'`

To make the test pass, I can use
```js
const input = screen.getByTestId('custom-input');
input.focus();
fireEvent.keyDown(input, {key: "h", code: "KeyH"});
fireEvent.change(input, { target: { value: "h" } });
expect(input.value).to.equal('h');
```
But this test does not catch the bug.


The last solution I have in mind is to add a listener to the root component to see if the default have been prevented or not. But with this solution we are not really testing if the input has been modified.

Reply:
I took a look into how to test this. You could add an e2e test to ensure that, when an input is rendered as the column header, typing into it will work. For the unit test, you could check if someone called `event.preventDefault` or not. The return value of `fireEvent.keyDown` is `false` when the event default action was cancelled by someone. It should be `true` when dispatched by a custom rendered column header. This is how `@testing-library/user-event` implements `userEvent.type`: https://github.com/testing-library/user-event/blob/a5ca2e47f30ea71b72a443c227ea125beb4e84b7/src/keyboard/keyboardAction.ts#L88